ffi/dotnet-idsdk/IoticsIdLib/src/AgentIdentity.cs

```diff
@@ -0,0 +1,9 @@
+namespace IOTICS;
+
+public class AgentIdentity(Identity factory, string seed, string keyName, string id, string did) : PartyIdentity(factory, seed, keyName, id, did)
+{
+    public void Recreate() 
```

Comment:
Why expose this? Is it a common use-case? (Upserting a document during the runtime of an agent as opposed to subsequent startup)

Reply:
recreate is exposed to recreate in the resolver the identity withouth having to manage removal of delegations etc.
TBH it's a utility method I have only used in demo code (not that I have written any production code recently)

Reply:
do you think it's confusing?

Reply:
No, I think having seen the rest it is OK. Though I suppose it would be nice (in the near future) for the FFI functions to more closely resemble the golang API. (And sure - maybe the latter should be revisited anyway.)